### ISSUE TITLE
Possible memory leak in src/chain.js

### ISSUE THREAD

**Issue by jacobbubu** [amark/gun#390]
https://github.com/amark/gun/blob/master/src/chain.js#L5

```
Gun.chain.chain = function(){
	var at = this._, chain = new this.constructor(this), cat = chain._;
	cat.root = root = at.root;                        // where does the root come from?
	cat.id = ++root._.once;
	cat.back = this;
```

**Comment by amark** [comment]
@jacobbubu thanks for seeing this! I edit/modify the built code directly, which has a `var root` in an outer scope, so this never got flagged as a global variable leak, but it indeed would/could be! I have fixed this in my unpublished `v0.8.x` branch, so expect this fixed when that gets published! Thanks for finding this!

What are you working on? What are you building? :) Would love to hear.

**Comment by jacobbubu** [comment]
I'm working on an E-commerce website and looking for a way to build up our next-generation server and storage architecture. Graph database comes to my research subject and a peer-2-peer decentralized graph db adds more  value on it.

**Comment by amark** [comment]
@jacobbubu nice! When you guys go big and become a massive mega-corporation with profits remember me ;) and sign up for our support plan! Have fun in the meanwhile, definitely ping on gitter  if you need any community help, and advertise yourself on our list-of-projects-using gun! Would love to see a demo when you have it. :D

**Comment by amark** [comment]
now fixed in `0.8.0` :) @jacobbubu thanks for reporting it. Closing.

@jacobbubu can I see a demo???

**Comment by jacobbubu** [comment]
For now, our store is just hosted on https://skinat.tmall.com. Build a self-hosted store with highly customizable functionalities is our dream.
